net/tcp/tcp_wrbuffer.c

```diff
@@ -105,6 +105,44 @@ void tcp_wrbuffer_initialize(void)
   nxsem_set_protocol(&g_wrbuffer.sem, SEM_PRIO_NONE);
 }
 
+/****************************************************************************
+ * Name: tcp_wrbuffer_inqueue_size
+ *
+ * Description:
+ *   Get the in-queued write buffer size from connection
+ *
+ * Input Parameters:
+ *   conn - The TCP connection of interest
+ *
+ * Assumptions:
+ *   Called from user logic with the network locked.
+ *
+ ****************************************************************************/
+
+uint32_t tcp_wrbuffer_inqueue_size(FAR struct tcp_conn_s *conn)
+{
+  FAR struct tcp_wrbuffer_s *wrb;
+  FAR sq_entry_t *entry;
+  uint32_t total = 0;
+
+  if (conn)
+    {
+      for (entry = sq_peek(&conn->unacked_q); entry; entry = sq_next(entry))
+        {
+          wrb = (FAR struct tcp_wrbuffer_s *)entry;
+          total += TCP_WBSENT(wrb);
+        }
+
+      for (entry = sq_peek(&conn->write_q); entry; entry = sq_next(entry))
+        {
+          wrb = (FAR struct tcp_wrbuffer_s *)entry;
+          total += TCP_WBSENT(wrb);
```

Comment:
TCP_WBSENT is on-wire bytes, which is not appropriate thing to look at for SO_SNDBUF.

i guess TCP_WBPKTLEN is more appropriate. how do you think?

Reply:
Done